dds/DCPS/RTPS/Util.cpp

```diff
@@ -0,0 +1,38 @@
+#include "Util.h"
+
+#include "dds/DCPS/Definitions.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace RTPS {
+
+bool bitmapNonEmpty(const OpenDDS::RTPS::SequenceNumberSet& snSet)
+{
+  const size_t num_ulongs = (snSet.numBits + 31) / 32;
+
+  OPENDDS_ASSERT(num_ulongs <= snSet.bitmap.length());
+
+  if (num_ulongs == 0) {
+    return false;
+  }
+
+  const size_t last_index = num_ulongs - 1;
+  if (last_index > 0) {
+    for (CORBA::ULong i = 0; i < last_index; ++i) {
+      if (snSet.bitmap[i]) {
+        return true;
+      }
+    }
+  }
+
+  const CORBA::ULong mod = snSet.numBits % 32;
+  const CORBA::ULong mask = mod ? ~((0x1u << (32 - mod)) - 1) : 0xFFFFFFFF;
```

Comment:
Another way (not sure it's better) to handle the special case of `mod == 0`.  The entire integer is used, so it could just be effectively `++last_index` in the loop above.

Reply:
I'm not entirely sure either. I think if we were optimizing for the mod == 0 case extending the loop would be ideal, but most of the time (probably much less than 1/32, since the majority of bitmaps will be short) we'll be dealing with a non-zero mod value. As it stands, I think I prefer the clarity of 1) loop & test for beginning ulongs 2) mask & test for final ulong.